### ISSUE TITLE
Unexpected results: IoU decreases when using visibility augmentation

### ISSUE THREAD

**Issue by Tauh8** [raphaelsulzer/convolutional_occupancy_networks#3]
## Description
I'm trying to reproduce the results from your paper "Deep Surface Reconstruction from Point Clouds with Visibility Information" with Convolutional Occupancy Networks. However, I'm getting results that contradict the paper's findings.

## Results I'm getting
- Baseline (without visibility): IoU = 0.92
- With AP only: IoU = 0.87
- With AP + SV: IoU = 0.84

This shows a decrease in performance when adding visibility information, which contradicts Table II in the paper where visibility information improves the IoU.

## Environment & Setup
- Dataset: ModelNet10 from your [link](https://zenodo.org/record/5940164#.YflZolvMLIE)
- Hardware: RTX4060

## Questions
1. Are there any specific hyperparameters that need to be tuned when using visibility information?
2. Could there be any preprocessing steps I might have missed for the visibility features?
3. Does the training process need to be modified when incorporating visibility information?

## Additional Information
My config:
method: conv_onet
data:
  dataset: ModelNet10
  classes: ["toilet","bed","table","chair","sofa","monitor","desk","nightstand","dresser","bathtub"]
  scan: 43
  input_type: pointcloud
  path: /home/tauh/ModelNet10
  pointcloud_n: null
  pointcloud_noise: null
  points_subsample: 2048
  points_file: points.npz
  points_iou_file: points.npz
  pointcloud_file: scan.npz
  voxels_file: null
  train_split: train
  val_split: test
  test_split: test
  dim: 3
sensor:
  mode: norm+
  vector: sensor
  stepsi: [-1]
  stepso: [ 1]
  sampling: uniform
  factor: neighborhood # or neighborhood
  ident_dim: 2
model:
  encoder: pointnet_local_pool
  encoder_kwargs:
    hidden_dim: 32
    plane_type: ['xz', 'xy', 'yz']
    plane_resolution: 64
    grid_resolution: 32
    unet: True
    unet_kwargs:
      depth: 4
      merge_mode: concat
      start_filts: 32
  decoder: simple_local
  decoder_kwargs:
    sample_mode: bilinear # bilinear / nearest
    hidden_size: 32
  c_dim: 32
training:
  out_dir: out/modelnet/2dAP
  load_model: best
  batch_size: 32
  model_selection_metric: iou
  model_selection_mode: maximize
  print_every: 200
  visualize_every: 100000000
  validate_every: 4000
  checkpoint_every: 20000
  backup_every: 20000
  n_workers: 4
  n_workers_val: 4
test:
  threshold: 0.2
  eval_mesh: true
  eval_pointcloud: false
generation:
  vis_n_outputs: 100000
  refine: false
  n_x: 128
  n_z: 1
  eval: ["iou"]

I would greatly appreciate any insights or guidance on this issue. Thank you!

![image](https://github.com/user-attachments/assets/601866b6-1299-485e-bd5a-a3c4ba9c89e9)
![image](https://github.com/user-attachments/assets/7a11900d-12cc-4eb1-a026-a4ff736667df)
![image](https://github.com/user-attachments/assets/ced7a405-4d39-4244-8b91-0c68184f6e99)
